### PR TITLE
Removing cookie banner from embedded projects

### PR DESCRIPTION
**Pull request by daynew**
> As a student, when I embed my Code.org project in my website, I don’t want the OneTrust banner to show. The banner doesn’t render well on small screens and blocks my project.

When a student creates a project, they can embed it in their personal website by appending `/embed` to their project URL.

For example https://studio.code.org/projects/flappy/WlhSpwQ_lHKTw67RI0Pu2uRh-XUsl7502OSGOk4YzD8/embed 

This removes the OneTrust banner so that students can have a better experience when sharing their projects.
## Screenshots
| before | after |
| --- | --- |
| Screenshot 2024-09-11 at 1 36 14 PM | Screenshot 2024-09-11 at 1 34 58 PM | 

## Links
* [JIRA](https://codedotorg.atlassian.net/browse/P20-1128)

## Testing story
* UI Tests
* Append `?otreset=true&otgeo=gb` to URLs in order to simulate the European cookie banner experience.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: shared/haml/onetrust_cookie_scripts.haml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 - require deploy_dir('shared/middleware/helpers/experiments')
 
+-# When student projects are embedded in other websites, the URL ends with
+-# /embed. When Code.org is rendered in an iframe on another website, we are
+-# responsible for getting user consent for Cookies and Tracking Technologies,
```
Comment (fisher-alice) [inline shared/haml/onetrust_cookie_scripts.haml:5]: Nit: we are 'not' responsible for getting user consent...

File: dashboard/test/ui/features/platform/one_trust.feature [hunk 2]
```diff
@@ -106,9 +105,27 @@ Feature: OneTrust integration
...
     | http://studio.code.org/users/sign_in                                    |
+
+  @as_student
+  Scenario Outline: Embedded projects do not display the OneTrust banner
+    Given I am on "<url>"
+    Then I switch to the embedded view of current project
+    Then I append "?otreset=true&otgeo=gb" to the URL
+    Then element "script[src$='otSDKStub.js']" does not exist
+    Then element "script[src$='OtAutoBlock.js']" does not exist
+  Examples:
```
Comment (fisher-alice) [inline dashboard/test/ui/features/platform/one_trust.feature:126]: Looks like a complete list!

**Review by fisher-alice — APPROVED** [review]
Thanks so much for taking care of this! ❤️

File: dashboard/test/ui/features/platform/one_trust.feature [hunk 3]
```diff
@@ -24,9 +23,9 @@ Feature: OneTrust integration
 
   Scenario: OneTrust cookie pop-up shows when self-hosting OneTrust libraries on code.org
     Given I create a student named "Alice"
-    Given I am on "http://studio.code.org/home?otreset=false&otgeo=gb"
+    Given I am on "http://studio.code.org/home?otreset=true&otgeo=gb"
```
Comment (juanmanzojr) [inline dashboard/test/ui/features/platform/one_trust.feature]: Was missing this in my original reverted PR, my new one adds this change: https://github.com/code-dot-org/code-dot-org/pull/61000

**Review by juanmanzojr — APPROVED** [review]
LGTM

**Review by wilkie — APPROVED** [review]
👍

**Pull request merged**